pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/reader/ForwardIndexReader.java

```diff
@@ -916,4 +919,67 @@ default byte[][] getBytesMV(int docId, T context) {
   default int getNumValuesMV(int docId, T context) {
     throw new UnsupportedOperationException();
   }
+
+  /**
+   * This class represents the buffer byte ranges accessed while reading a given docId.
+   * See {@link ValueRangeProvider} for more details.
+   */
+  @AllArgsConstructor
+  @EqualsAndHashCode
+  @Getter
+  class ValueRange {
+    private final long _offset;
+    private final int _sizeInBytes;
+
+    public static ValueRange newByteRange(long offset, int sizeInBytes) {
```

Comment:
Is this useful? This is the same as the constructor

Reply:
Earlier we had a few variants here. Now it is indeed same as the constructor. Removed.